### PR TITLE
Fix CSV Test Cases

### PR DESCRIPTION
**Pull request by davidcui-amzn**
*Issue #, if available:*
N/A
*Description of changes:*
Fixed test cases to run successfully 
* Automatically creates `datareport` index and adds docs needed to pass test cases 
* Fix ES id for test cases 
* Add JSON data for the docs to be added to the `datareport` index
* renamed `report.ts` to `visualReport.ts` to differentiate with `dataReport`

Current pre-requisites for test cases to pass:
* Elasticsearch must be running on `localhost:9200`
* Kibana must be running on `localhost:5601`
* The `.kibana` index must have the `kibana_sample_data_flights` sample data installed 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: server/routes/utils/__tests__/dataReportTestDataLarge.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (zhongnansu) [inline server/routes/utils/__tests__/dataReportTestDataLarge.json:1]: it's better to put test data(.json) in a folder

File: server/routes/utils/dataReportHelpers.ts [hunk 2]
```diff
@@ -117,14 +117,13 @@ export const buildQuery = (report, is_count) => {
...
-
   if (report._source.timeFieldName && report._source.timeFieldName.length > 0) {
     requestBody.must(
       esb
         .rangeQuery(report._source.timeFieldName)
         .format('epoch_millis')
-        .gte(report._source.start)
-        .lte(report._source.end)
+        .gte(report._source.start - 1)
+        .lte(report._source.end + 1)
```
Comment (zhongnansu) [inline server/routes/utils/dataReportHelpers.ts:126]: why is this hardcoded?

File: server/routes/index.ts [hunk 3]
```diff
@@ -13,17 +13,17 @@
...
 import registerDataReportMetadata from './dataReportMetadata';
 import dashboardRoute from './getDashboards';
 import { IRouter } from '../../../../src/core/server';
 
 export default function (router: IRouter) {
-	registerReportRoute(router);
-	registerReportDefinitionRoute(router);
-	registerDataReportMetadata(router);
-	registerDataReport(router);
+  registervisualReportRoute(router);
```
Comment (akbhatta) [inline server/routes/index.ts:24]: nit: registervisualReportRoute --> registerVisualReportRoute

File: public/components/report_definitions/report_settings/report_settings.tsx [hunk 4]
```diff
@@ -398,7 +398,7 @@ export function ReportSettings(props) {
     let baseUrl = window.location.href;
     return baseUrl.replace(
       'opendistro_kibana_reports#/create',
-      'dashboards#/view/'
+      'kibana#/dashboard/'
```
[the remark this answers is not in the document]
Comment (davidcui-amzn) [reply public/components/report_definitions/report_settings/report_settings.tsx:401]: The url for dashboards in Kibana 7.8 is different from 8.0

File: server/routes/index.ts [hunk 5]
```diff
@@ -13,17 +13,17 @@
...
+  registerReportDefinitionRoute(router);
+  registerDataReportMetadata(router);
+  registerDataReport(router);
   dashboardRoute(router);
```
[the remark this answers is not in the document]
Comment (davidcui-amzn) [reply server/routes/index.ts:28]: Good call- will change, maintain consistency

File: server/routes/utils/__tests__/dataReportTestDataLarge.json (hunk 1 above)
Comment (davidcui-amzn) [reply server/routes/utils/__tests__/dataReportTestDataLarge.json:1]: Sure, will add to a `test_data` folder

File: server/routes/utils/dataReportHelpers.ts (hunk 2 above)
Comment (davidcui-amzn) [reply server/routes/utils/dataReportHelpers.ts:126]: Test cases failed

**Pull request merged**